core/src/main/java/org/aerogear/mobile/core/ServiceModuleRegistry.java

```diff
@@ -0,0 +1,27 @@
+package org.aerogear.mobile.core;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Registry of service modules.  It is responsible for maintaining a mapping of service modules to
+ * their types as well as the dependencies each module expects.
+ */
+public final class ServiceModuleRegistry {
+    private static final Map<String, Class<? extends ServiceModule>> serviceTypeMap = new HashMap<>();
```

Comment:
How this will be used?
Will this work assuming that namespace can have only one service of certain type.?
It may be worth to check if that is not limitation for some of the planned services.